tests/flytekit/unit/types/directory/test_dir.py

```diff
@@ -1,52 +1,75 @@
+import tempfile
 from pathlib import Path
 from typing import Optional
 
+import pytest
 import flytekit
 from flytekit import task, workflow
 from flytekit.types.directory import FlyteDirectory
 
 
-def test_src_path_with_different_types() -> None:
-    N_FILES = 3
+N_FILES_PER_DIR = 3
+
+@pytest.fixture
+def local_tmp_dirs():
+    # Create a source directory
+    src_dir = tempfile.TemporaryDirectory()
+    for file_idx in range(N_FILES_PER_DIR):
+        with open(Path(src_dir.name) / f"{file_idx}.txt", "w") as f:
+            f.write(str(file_idx))
+
+    # Create an empty directory as the destination
+    dst_dir = tempfile.TemporaryDirectory()
+
+    yield src_dir.name, dst_dir.name
+
+    # Cleanup
+    src_dir.cleanup()
+    dst_dir.cleanup()
```

Comment:
Can use use a context manager here?

```suggestion
    # Create a source directory
    with (tempfile.TemporaryDirectory() as src_dir,
          tempfile.TemporaryDirectory() as dst_dir):
        for file_idx in range(N_FILES_PER_DIR):
            with open(Path(src_dir) / f"{file_idx}.txt", "w") as f:
                f.write(str(file_idx))

        yield src_dir, dst_dir
```

Reply:
sorry I just merge it and didn't see this, @arbaobao can you do this when you back from work?

Reply:
Let me deal with it!

Hi @thomasjpfan, may I ask if it's possible to push the fixed version here? Or, is it better to fix it in another PR?

Reply:
fix in another PR